megatron/model/gpt2_model.py

```diff
@@ -45,7 +45,11 @@
 
 
 def gpt2_attention_mask_func(attention_scores, ltor_mask):
-    attention_scores.masked_fill_(ltor_mask, -10000.0)
+    mask_value = torch.finfo(attention_scores.dtype).min
+    # Need to be a tensor, otherwise we get error: `RuntimeError: expected scalar type float but found double`.
+    # Need to be on the same device, otherwise `RuntimeError: ..., x and y to be on the same device`
+    mask_value = torch.tensor(mask_value, dtype=attention_scores.dtype).to(attention_scores.device)
```

Comment:
It's always best to initialize tensors directly on the target device. Using `.to` necessitates a copy first on the torch default device (`'cpu'` unless explicitly set).

Change to:
`mask_value = torch.tensor(mask_value, dtype=attention_scores.dtype, device=attention_scores.device)`

Reply:
Sounds good, fixed below